### PR TITLE
Nested types for a given type

### PR DESCRIPTION
**Pull request by andredered**
Added solution for https://github.com/TNG/ArchUnitNET/issues/179#issue-1378631024

### CODE REVIEW

File: ArchUnitNET/Fluent/Syntax/Elements/Types/TypeConditionsDefinition.cs [hunk 1]
```diff
@@ -267,6 +267,183 @@ IEnumerable<ConditionResult> Condition(IEnumerable<TRuleType> ruleTypes, Archite
...
+
+            return new EnumerableCondition<TRuleType>(Condition, description);
+        }
+
+        public static ICondition<TRuleType> BeNestedIn(IEnumerable<Type> types)
+        {
+            var typeList = types.ToList();
+            var firstType = typeList.First();
+
+            IEnumerable<ConditionResult> Condition(IEnumerable<TRuleType> ruleTypes, Architecture architecture)
```
Comment (MSinstein) [inline ArchUnitNET/Fluent/Syntax/Elements/Types/TypeConditionsDefinition.cs:404]: I would suggest to extract the common parts of this method and the `Condition` method above.

**Review by MSinstein — COMMENTED** [review]
I would prefer the code to be much more DRY. Due to several repetitions of similar code, some of the comments also apply to several places. Furthermore, it would be good to have at least some tests that cover the new functionality, e.g. like there are for `AreAssignableTo`

File: ExampleTest/ExampleArchUnitTest.cs [hunk 2]
```diff
@@ -53,7 +53,6 @@ public void TypesShouldBeInCorrectLayer()
             //check if your architecture fulfils your rules
             exampleClassesShouldBeInExampleLayer.Check(Architecture);
             forbiddenInterfacesShouldBeInForbiddenLayer.Check(Architecture);
-
```
Comment (MSinstein) [inline ExampleTest/ExampleArchUnitTest.cs:56]: I'd keep the newline

**Review by MSinstein — COMMENTED** [review]
I would be fine if de-duplications in `TypeConditionsDefinition` are done in a separate PR

**Review by fgather — APPROVED** [review]

**Pull request merged**